docs/DeveloperGuide.md

```diff
@@ -294,17 +294,71 @@ The activity diagram shows the general sequence of steps when a user interacts w
 
 <br>
 
+## Implementation of Entity Commands
+Entity commands include `add`, `delete`, `find`, `clear` commands
+
+**Step 1**. The user types an `xyz` command in the `CommandBox`, followed by the type of entity `person` or `appt`, followed by appropriate arguments and prefixes.
+**Step 2**. The command is passed to the `LogicManager`. `LogicManager` then calls the `AddressBookParser::parseCommand` method to parse the command
+**Step 3**. The `AddressBookParser` then creates a `xyzEntityCommand` by passing string to `xyzCommandParser` and is returned to the `LogicManager`.
+**Step 4**. The `LogicManager` calls the `xyzCommand : execute` method which creates a `CommandResult` Object.
+**Step 5**. The `CommandResult` object is returned to the `LogicManager`.
+
 ### Find appointment feature
 
 #### Implementation
+When a user types a `find appt` command, the DocTrack application will 
+find the appointment based on the parameters provided.
+
+**Step 1**. The user types the `find appt` command in the `CommandBox` followed by the name or a date
+The date is specified using a `d/` prefix and the name is specified using `n/` prefix.
+
+**Step 2**. The command is passed to the `LogicManager`. `LogicManager` then calls the `AddressBookParser::parseCommand`
+method to parse the `find appt` command
+
+**Step 3**. The `AddressBookParser` creates a `FindAppointmentCommand` by passing the string to `FindCommandParser` 
+and is then returned to the `LogicManager`.
+
+**Step 4**. The `LogicManager` calls the `FindAppointmentCommand:execute` method which creates a new `CommandResult` Object
+
+**Step 5**. The `CommandResult` object is returned to the `LogicManager`.
 
 #### Design considerations
+**Aspect: How to show find appointment.**
+
+- **Alternative 1 (Current choice)**: Find the information based on what the user has provided (name, date).
+  - Pros: Fast and easy to find by date and name
+  - Cons: Confusing syntax from user's perspective
+
+- **Alternative 2**: Create different find commands, find by date, find by name etc.
+  - Pros: Much easy in terms of user experience
+  - Cons: Harder to implement as more code needs to be written.
 
 <br>
 
 ### List appointment feature
 
 #### Implementation
+When a user types a `list appt` command, the DocTrack application will
+return a list of all the appointments in the appointment book.
+
+**Step 1**. The user types the `list` command in the `CommandBox`. The command is passed to the `LogicManager`. 
+
+**Step 2**. `LogicManager` then calls the `AddressBookParser::parseCommand`
+method to parse the `list appt` command
+
+**Step 3**. The `AddressBookParser` creates a `ListAppointmentCommand` by passing the string to `ListCommandParser`
+and is then returned to the `LogicManager`.
+
+**Step 4**. The `LogicManager` calls the `ListAppointmentCommand:execute` method which creates a new `CommandResult` Object
+
+**Step 5**. The `CommandResult` object is returned to the `LogicManager`.
```

Comment:
Should we remove these implementation steps, since they would already be described by the general sequence of steps above for `xyz` command?